Base/Python/slicer/parameterNodeWrapper/serializers.py

```diff
@@ -970,3 +966,50 @@ def supportsCaching(self):
         not update the underlying parameter node, which can be confusing.
         """
         return all([s.supportsCaching() for s in self._serializers])
+
+
+@parameterNodeSerializer
+class EnumSerializer(Serializer):
+    @staticmethod
+    def canSerialize(type_) -> bool:
+        return issubclass(type_, enum.Enum)
+
+    @staticmethod
+    def create(type_):
+        return EnumSerializer(type_)
+
+    def __init__(self, enum_):
+        self.enum = enum_
+
+        if len(self.enum) == 0:
+            raise ValueError('Cannot serialize empty Enum.')
+
+    def default(self):
+        # There is no idiomatic way to get a default value of an arbitrary enum,
+        # so just take the first item in iteration order.
+        return next(iter(self.enum))
+
+    def isIn(self, parameterNode, name: str) -> bool:
+        return parameterNode.HasParameter(name)
+
+    def write(self, parameterNode, name: str, value) -> None:
+        if not isinstance(value, self.enum):
```

Comment:
`write` should assume the value is valid. The `create` method should look like

```py
def create(type_):
    if EnumSerializer.canSerialize(type_):
        return ValidatedSerializer(EnumSerializer(type_),
                                   [IsInstance(type_)])
    return None
```

so the `isinstance` is a validator that can be used by things like the `UnionSerializer`.

Reply:
Wondering if it would also be appropriate to include `Choice(list(type_))` here? It seems appropriate but I'm not entirely certain if there are other implications of this.

Reply:
Agree that adding `Choice` is redundant and not necessary.